src/main/routing/shd-topology.c

```diff
@@ -122,6 +122,54 @@ static gboolean _topology_loadGraph(Topology* top, const gchar* graphPath) {
     return TRUE;
 }
 
+/** Returns FALSE if issue parsing graph, otherwise returns TRUE.
+ * If returning TRUE, whether or not the graph is complete is stored in result.
+ */
+static gboolean _topology_isComplete(Topology* top, gboolean *result) {
+    igraph_t *graph = &top->graph;
+    igraph_vs_t vs;
+    igraph_vit_t vit;
+    int ret;
+    long int vcount = igraph_vcount(graph);
+    /*
+     * Determines if a graph is complete by:
+     * - knowning how many vertexes there are
+     * - for each vertex, count the indcident edges
+     *   - if less than the number of vertexes, it isn't a complete graph
+     * - otherwise the graph is complete
+    /* vert selector. We wall all verts */
+    ret = igraph_vs_all(&vs);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vs_all returned non-success code %i", ret);
+        return FALSE;
+    }
+    ret = igraph_vit_create(graph, vs, &vit);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vit_create returned non-success code %i", ret);
+        return FALSE;
+    }
+    while (!IGRAPH_VIT_END(vit)) {
+        igraph_vector_t iedges;
+        igraph_vector_init(&iedges, 0);
+        ret = igraph_incident(graph, &iedges, IGRAPH_VIT_GET(vit), IGRAPH_OUT);
+        if (ret != IGRAPH_SUCCESS) {
+            critical("error computing igraph_incident\n");
+            return FALSE;
+        }
+        long int ecount = igraph_vector_size(&iedges);
+        if (ecount < vcount) {
+            info("Vert id=%li has %li incident edges to %li total verts "
+                "and thus this isn't a complete graph",
+                IGRAPH_VIT_GET(vit), ecount, vcount);
+            *result = FALSE;
+            return TRUE;
+        }
+        IGRAPH_VIT_NEXT(vit);
+    }
+    *result = TRUE;
```

Comment:
Please add a check that the `result` pointer is non-null before dereferencing it.